### DISCUSSION TITLE
installation

### DISCUSSION THREAD

**Discussion by mjstrens** [pyvista/pyvista#5518] [Q&A] [answered]
this command is given in the docs:
pip install 'jupyterlab>=3' ipywidgets 'pyvista[all,trame]'

but I get this error:
ERROR: Invalid requirement: "'jupyterlab"

what am I doing wrong, please?

It's a conda env on Windows

**Comment by MatthewFlamm** [comment]
Note this part includes a single quote at the beginning. `"'jupyterlab"` in 
> ERROR: Invalid requirement: "'jupyterlab"

In windows, you need to use double quotes only.  If you can, would you raise a pull request to fix this or point out where this is in the docs?

**Reply by mjstrens** [reply, marked as answer]
Thanks - the command works with the single quotes replaced with double quotes. The single quote in the error message just shows the windows anaconda command line couldn't parse my input (given above). I won't attempt a pull request as I'm not familiar with doing this on 'proper' projects.

````
pip install "jupyterlab>=3" ipywidgets "pyvista[all,trame]"
````
Unfortunately, pyvista still does not work for me with the trame backend - I just get a blank window with no content. But it does work if I use 'static' for the backend. Code copied below - but that would be a different issue so I think this one is closed.

````
import pyvista as pv
import numpy as np

pv.set_jupyter_backend('trame')  

def draw_beam(length, width, height):
    # Create a rectangular prism (box) to represent the beam
    beam = pv.Box(bounds=(0, length, 0, width, 0, height))
    # Create a plotter object and add the beam to it
    plotter = pv.Plotter()
    plotter.add_mesh(beam, color='burlywood')
    # Display the plot
    plotter.show()

# Call the function with dimensions of the beam
draw_beam(10, 2, 2)
````
